lib/std/json/hashmap.zig

```diff
@@ -27,11 +27,12 @@ pub fn ArrayHashMap(comptime T: type) type {
 
             if (.object_begin != try source.next()) return error.UnexpectedToken;
             while (true) {
-                const token = try source.nextAlloc(allocator, .alloc_if_needed);
+                const token = try source.nextAlloc(allocator, options.allocate.?);
                 switch (token) {
                     inline .string, .allocated_string => |k| {
                         const gop = try map.getOrPut(allocator, k);
-                        if (token == .allocated_string) {
+                        // Don't free the key if its allocation was intentional
+                        if (token == .allocated_string and options.allocate != .alloc_always) {
```

Comment:
i think we actually just want to never free the keys of a hashmap, because accessing the string data is required for normal operation of a hash map.